packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m

```diff
@@ -40,6 +40,99 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
       NSURL *url = [[NSBundle mainBundle] URLForResource:name withExtension:extension];
       result(url.absoluteString);
       return;
+    } else if ([@"loadSystemImage" isEqualToString:call.method]) {
+      if (@available(iOS 13, *)) {
+        NSString *name = call.arguments[0];
+        NSNumber *pixelSizeWithDouble = call.arguments[1];
+        // iOS adds 15% padding to the outside of the image so we scale down to match the requested
+        // size.
+        double pointSize = [pixelSizeWithDouble doubleValue] * 0.85;
+        NSNumber *weightIndex = call.arguments[2];
+
+        // Up to 3 rgb values for primary, seconday and tertiary colors.
+        // see
+        // https://developer.apple.com/documentation/uikit/uiimagesymbolconfiguration/3810054-configurationwithpalettecolors
+        NSArray *rgbaValuesList = call.arguments[3];
+
+        NSMutableArray *colorArray = [[NSMutableArray alloc] init];
+
+        for (int i = 0; i < [rgbaValuesList count]; i += 4) {
+          UIColor *primaryColor = [UIColor colorWithRed:[rgbaValuesList[i] doubleValue]
+                                                  green:[rgbaValuesList[i + 1] doubleValue]
+                                                   blue:[rgbaValuesList[i + 2] doubleValue]
+                                                  alpha:[rgbaValuesList[i + 3] doubleValue]];
+          [colorArray addObject:primaryColor];
+        }
+
+        UIImageSymbolWeight weight = UIImageSymbolWeightRegular;
+
+        switch ([weightIndex integerValue]) {
+          case 0:
+            weight = UIImageSymbolWeightUltraLight;
+            break;
+          case 1:
+            weight = UIImageSymbolWeightThin;
+            break;
+          case 2:
+            weight = UIImageSymbolWeightLight;
+            break;
+          // 3 is regular
+          case 4:
+            weight = UIImageSymbolWeightMedium;
+            break;
+          case 5:
+            weight = UIImageSymbolWeightSemibold;
+            break;
+          case 6:
+            weight = UIImageSymbolWeightBold;
+            break;
+          case 7:
+            weight = UIImageSymbolWeightHeavy;
+            break;
+          case 8:
+            weight = UIImageSymbolWeightBlack;
+            break;
+          default:
+            weight = UIImageSymbolWeightRegular;
+            break;
+        }
+
+        UIImageSymbolConfiguration *pointSizeConfig =
+            [UIImageSymbolConfiguration configurationWithPointSize:pointSize
+                                                            weight:weight
+                                                             scale:UIImageSymbolScaleDefault];
+
+        UIImage *finalImage;
+
+        if (@available(iOS 15, *)) {
+          NSNumber *preferMulticolor = call.arguments[4];
+          UIImageSymbolConfiguration *colors;
```

Comment:
You can initialize this inline with a conditional so it's clear that it isn't ever nil:

```
UIImageSymbolConfiguration *colors = [preferMulticolor boolValue]
    ? ...
    : ...;
```